packages/core/saved-objects/core-saved-objects-base-server-internal/src/serialization/serializer.test.ts

```diff
@@ -7,28 +7,47 @@
  */
 
 import _ from 'lodash';
-import type { SavedObjectsRawDoc } from '@kbn/core-saved-objects-server';
+import type { SavedObjectsRawDoc, ISavedObjectTypeRegistry } from '@kbn/core-saved-objects-server';
 import { SavedObjectsSerializer } from './serializer';
-import { typeRegistryMock } from '../saved_objects_type_registry.mock';
 import { encodeVersion } from '../version';
-import { LEGACY_URL_ALIAS_TYPE } from '../object_types';
+import { LEGACY_URL_ALIAS_TYPE } from '../legacy_alias';
+
+const createMockedTypeRegistry = ({
+  isNamespaceAgnostic,
+  isSingleNamespace,
+  isMultiNamespace,
+}: {
+  isNamespaceAgnostic: boolean;
+  isSingleNamespace: boolean;
+  isMultiNamespace: boolean;
+}): ISavedObjectTypeRegistry => {
+  const typeRegistry: Partial<ISavedObjectTypeRegistry> = {
+    isNamespaceAgnostic: jest.fn().mockReturnValue(isNamespaceAgnostic),
+    isSingleNamespace: jest.fn().mockReturnValue(isSingleNamespace),
+    isMultiNamespace: jest.fn().mockReturnValue(isMultiNamespace),
+  };
+  return typeRegistry as ISavedObjectTypeRegistry;
+};
```

Comment:
Only test of the `@kbn/core-saved-objects-base-server-internal` that was using a mock from code that was moved to this package. I KISS'ed and created an inline mock helper. 

The alternative would have been to create 2 additional packages (impl+mock) just for the SO type registry, and it didn't feel worth it.